### PR TITLE
feat: add origin selector

### PR DESCRIPTION
**Pull request by lfjnascimento**
- add Select component from shadcn
- add Select component which is a custom version of ui/Select
- add origin filter to tree endpoint
- add origin selector to top bar

image

close: #78

### CODE REVIEW

File: dashboard/src/components/Select/Select.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  SelectValue,
+} from '@/components/ui/select';
+
+const Select = (props: React.ComponentProps<typeof SelectUI>): JSX.Element => {
+  const { children, ...propsWithoutChildren } = props;
+
+  return (
+    <SelectUI {...propsWithoutChildren}>
+      <SelectTrigger className="w-auto rounded-full border-2 border-dimGray px-6 py-4 text-base font-medium text-dimGray">
+        <SelectValue placeholder="Theme" />
```
Comment (mari1912) [inline dashboard/src/components/Select/Select.tsx:15]: should we add a better placeholder and make it use i18n?

File: dashboard/src/components/TopBar/TopBar.tsx [hunk 2]
```diff
@@ -1,12 +1,62 @@
...
+  useEffect(() => {
+    if (unsafeOrigin === undefined)
+      navigate({
+        search: previousSearch => ({ ...previousSearch, origin: origin }),
+      });
+  });
+
+  return (
+    <div className="flex items-center">
+      <span className="mr-4 text-base font-medium text-dimGray">Origin</span>
```
Comment (mari1912) [inline dashboard/src/components/TopBar/TopBar.tsx:44]: use intl

File: dashboard/src/components/TopBar/TopBar.tsx [hunk 3]
```diff
@@ -1,12 +1,62 @@
...
+  );
+};
+
 const TopBar = (): JSX.Element => {
   return (
     <div className="fixed top-0 z-10 mx-52 flex h-20 w-full bg-white pl-6 pr-12">
-      <div className="flex w-full flex-row items-center justify-between">
-        <span className="text-2xl">
+      <div className="flex flex-row items-center justify-between">
+        <span className="mr-14 text-2xl">
```
Comment (mari1912) [inline dashboard/src/components/TopBar/TopBar.tsx:56]: just make sure it will be aligned with other content, because of the changes we are making

**Review by mari1912 — COMMENTED** [review]
It's working nice. I just want to make sure that it is expected to be able to change the origin in any place and when we do it, go back to the revisions initial page, right?

File: dashboard/src/components/Select/Select.tsx (hunk 1 above)
Comment (lfjnascimento) [reply dashboard/src/components/Select/Select.tsx:15]: actually this shouldn't be here, i was using it for testing purposes. Nice catch

Comment (mari1912) [reply dashboard/src/components/Select/Select.tsx:15]: can we remove it?

**Review by mari1912 — APPROVED** [review]

**Pull request closed without merging**